crates/ty_python_semantic/resources/mdtest/type_compendium/never.md

```diff
@@ -0,0 +1,169 @@
+```
+
+## `Never` in annotations
+
+`Never` can be used in functions to indicate that the function never returns. For example, if a
+function always raises an exception, if it calls `sys.exit()`, if it enters an infinite loop, or if
+it calls itself recursively. All of these functions "Never" return control back to the caller:
+
+```py
+from typing_extensions import Never
+
+def raises_unconditionally() -> Never:
+    raise Exception("This function always raises an exception")
+
+def exits_unconditionally() -> Never:
+    import sys
+
+    return sys.exit(1)
+
+def loops_forever() -> Never:
+    while True:
+        pass
+
+def recursive_never() -> Never:
+    return recursive_never()
+```
+
+Similarly, if `Never` is used in parameter positions, it indicates that the function can "Never" be
+called, because it can never be passed a value of type `Never` (there are none):
+
+```py
+def can_not_be_called(n: Never) -> int:
+    return 0
+```
+
+## `Never` is disjoint from every other type
+
+Two types `A` and `B` are disjoint if their intersection is empty. Since `Never` has no inhabitants,
+it is disjoint from every other type:
+
+```py
+from ty_extensions import static_assert, is_disjoint_from
+from typing_extensions import Never
+
+class C: ...
+
+static_assert(is_disjoint_from(Never, int))
+static_assert(is_disjoint_from(Never, object))
+static_assert(is_disjoint_from(Never, C))
+static_assert(is_disjoint_from(Never, Never))
+```
+
+## Unions with `Never`
+
+`Never` can always be removed from unions:
+
+```py
+from ty_extensions import static_assert, is_equivalent_to
+from typing_extensions import Never
+
+static_assert(is_equivalent_to(int | Never | str | None, int | str | None))
+```
+
+## Intersections with `Never`
+
+Intersecting with `Never` results in `Never`:
+
+```py
+from ty_extensions import static_assert, is_equivalent_to, Intersection
+from typing_extensions import Never
+
+static_assert(is_equivalent_to(int | Never | str | None, int | str | None))
+```
+
+## `Never` is the complement of `object`
+
+`object` describes the set of all possible values, while `Never` describes the empty set. The two
+types are complements of each other:
+
+```py
+from ty_extensions import static_assert, is_equivalent_to, Not
+from typing_extensions import Never
+
+static_assert(is_equivalent_to(Not[object], Never))
+static_assert(is_equivalent_to(Not[Never], object))
+```
+
+This duality is also reflected in other facts:
+
+- `Never` is a subtype of every type, while `object` is a supertype of every type.
+- `Never` is assignable to every type, while `object` is assignable from every type.
+- `Never` is disjoint from every type, while `object` overlaps with every type.
+- Building a union with `Never` is a no-op, intersecting with `object` is a no-op.
+- Interecting with `Never` results in `Never`, building a union with `object` results in `object`.
+
+## Lists of `Never`
+
+`list[Never]` is a reasonable type. It has one inhabitant, the empty list:
```

Comment:
nit: "one inhabitant" feels slightly imprecise here, because it's possible to create multiple empty lists which are not the same runtime object but would all inhabit the type `list[Never]`